docs/devGuide/writingComponents.md

```diff
@@ -0,0 +1,115 @@
+{% set title = "Writing Components" %}
+<span id="title" class="d-none">{{ title }}</span>
+
+<frontmatter>
+  title: "{{ title }}"
+  layout: devGuide.md
+  pageNav: default
+</frontmatter>
+
+# {{ title }}
+
+<div class="lead">
+
+This page explains various concerns related to MarkBind components, focused on implementation and testing.
+</div>
+
+## Implementing Components
+
+There are multiple ways to implement MarkBind components.
+
+<box type='warning' light>
+
+If an author has a conflicting slot and attribute, MarkBind should **log a warning** to let them know!
+</box>
+
+### Transforming the Node Directly
+
+One way to implement a MarkBind component is to transform the node itself.
+This is a more low-level implementation that can be useful when a node only needs to be modified slightly.
+
+When a node is processed, MarkBind syntax is converted to HTML, and any remaining attributes will also be converted to HTML attributes. 
+This can be useful if you just need to add a HTML attribute to the node, or modify the value of an existing attribute.
+
+These transformations may take place at various stages of node processing: before (`preProcessNode`), during (`processNode`), or after (`postProcessNode`).
+
+{{ icon_examples }} 
+* Adding a class to a node (setting line numbers for code blocks)
+* Modifying attributes and adding a directive to a node ([former implementation](https://github.com/MarkBind/markbind/blob/502df135e07baebd9d4eea8ccc0654c990047792/packages/core/src/html/bootstrapVueProcessor.js#L73) of popovers)
+
+</box>
+
+### Vue Components
+
+Many MarkBind components are implemented as Vue components, either by creating a component in the `vue-components` package, or by importing a component from an external library.
+This can be useful when a more complicated set of features is needed, where a Vue component can provide an interface for us to manage these functionalities.
+
+Vue components are registered in `vue-components/src/index.js`, which allows them to be used in the template section of any Vue instance without needing to be imported first.
+
+<panel header="How do MarkBind attributes/slots get passed to the Vue component?">
+
+##### Attributes
```

Comment:
I think we can include the same tooltip information here as you did below; the reader won't see the bottom definition until after they read this if they read chronologically